src/js/lib/tokenizer.ts

```diff
@@ -0,0 +1,186 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+import { Tensor } from "openvino-node";
+import { ChatHistory } from "./addon.js";
+
+/**
+ * TokenizedInputs contains input_ids and attention_mask tensors.
+ * This is the result of encoding prompts using the Tokenizer.
+ */
+export interface TokenizedInputs {
+  /** Tensor containing token IDs for the encoded input */
+  input_ids: Tensor;
+  /** Tensor containing attention mask (1 for real tokens, 0 for padding) */
+  attention_mask: Tensor;
+}
+
+/**
+ * Options for encode method.
+ */
+export interface EncodeOptions {
+  /**
+   * Whether to add special tokens like BOS, EOS, PAD.
+   * @defaultValue true
+   */
+  addSpecialTokens?: boolean;
+
+  /**
+   * Whether to pad the sequence to the maximum length.
+   * @defaultValue false
+   */
+  padToMaxLength?: boolean;
+
+  /**
+   * Maximum length of the sequence.
+   * If undefined, the value will be taken from the IR.
+   */
+  maxLength?: number;
+
+  /**
+   * Side to pad the sequence, can be 'left' or 'right'.
+   * If undefined, the value will be taken from the IR.
+   */
+  paddingSide?: "left" | "right";
```

Comment:
Can be None, which means that the default padding side (which is stored in IR) is used.

Reply:
PaddingSide can be undefined, in which case the default value from the core part is used.